### PR TITLE
doc/md: add procedure testing example and doc

### PR DESCRIPTION
**Pull request by hilakashai**

### CODE REVIEW

File: doc/md/guides/testing/testing-procedures.mdx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  END;
+  SQL
+}
+// highlight-end
+```
+
+In the schema above we have a simple `users` table and a `check_user_email` procedure. The procedure checks whether or not
+the `email` column in the `users` table has a valid email format.
+
+### Project File
```
Comment (a8m) [inline doc/md/guides/testing/testing-procedures.mdx:72]:
```suggestion
### Config File
```

File: doc/md/guides/testing/testing-procedures.mdx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+Before we begin testing, create a [project file](/atlas-schema/projects#project-files) named
```
Comment (a8m) [inline doc/md/guides/testing/testing-procedures.mdx:74]:
```suggestion
Before we begin testing, create a [config file](/atlas-schema/projects#project-files) named
```

File: doc/md/guides/testing/testing-procedures.mdx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+```hcl title="schema.test.hcl"
+test "schema" "procedure" {
+  # Attempt to create user with invalid email (missing @)
+  catch {
+    sql = "CALL check_user_email('new_user', 'invalidemail')"
+  }
+  log {
+    message = "Invalid email passed"
+  }
```
Comment (a8m) [inline doc/md/guides/testing/testing-procedures.mdx:111]: better to add another `exec` block that query the affects and compare the result.

File: doc/md/guides/testing/testing-procedures.mdx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+  # Additional test case (optional)
+  exec {
+    sql = "CALL check_user_email('another_user', 'valid@email.com')"
+  }
+  log {
+    message = "Valid email passed"
```
Comment (a8m) [inline doc/md/guides/testing/testing-procedures.mdx:118]: ditto

File: doc/website/sidebars.js [hunk 5]
```diff
@@ -288,6 +288,10 @@ module.exports = {
                     type: 'doc',
                     id: 'guides/testing/testing-domains',
                     label: 'Testing Domains'
+                },                {
```
Comment (a8m) [inline doc/website/sidebars.js:291]:
```suggestion
                },
                {
```

File: doc/md/atlas-schema/hcl.mdx [hunk 6]
```diff
@@ -1700,6 +1700,75 @@ procedure "p8" {
 </TabItem>
 </Tabs>
 
+
+:::info <a class="sticky-anchor" id={"#testing-procedures"} href={"#testing-procedures"}>Testing Procedures</a>
+
+Atlas testing framework allows you to write unit tests for your procedures. The following example demonstrates how
```
Comment (rotemtam) [inline doc/md/atlas-schema/hcl.mdx:1706]: the atlas testing framework
atlas's testing framework

File: doc/md/atlas-schema/hcl.mdx [hunk 7]
```diff
@@ -1700,6 +1700,75 @@ procedure "p8" {
...
+test "schema" "procedure" {
+  # Attempt to create user with invalid email (missing @)
+  catch {
+    sql = "CALL check_user_email('new_user', 'invalidemail')"
+  }
+  log {
+    message = "Invalid email passed"
+  }
+
+# Additional test case (optional)
```
Comment (rotemtam) [inline doc/md/atlas-schema/hcl.mdx:1723]:
```suggestion
  # Additional test case (optional)
```

**Review by rotemtam — APPROVED** [review]

**Pull request merged**